arangod/Aql/OptimizerRules.cpp

```diff
@@ -2867,14 +2882,29 @@ void arangodb::aql::removeUnnecessaryCalculationsRule(Optimizer* opt,
           // expression types (V8 vs. non-V8) do not match. give up
           continue;
         }
-
-        if (!n->isInInnerLoop() && rootNode->callsFunction() && other->isInInnerLoop()) {
-          // original expression calls a function and is not contained in a loop
-          // we're about to move this expression into a loop, but we don't want
-          // to move (expensive) function calls into loops
-          continue;
+        
+        if (rootNode->callsFunction() && other->isInInnerLoop()) {
+          if (n->getLoop() != other->getLoop()) {
```

Comment:
(This is micro optimization)
I am actually not convinced that this matters as soon as you read one Document, just for the sake of completeness.

getLoop() is implemented using a "walker" over dependencies, this potentially has some computation time involved.
Here we need n->getLoop() twice. and looking at isInInnerLoop() impleentation of other we also need the loop of other twice.
So we could save some lookups if we fetch both before their ifs.

I am also not sure if compiler is clever enough to figure out that n->getLoop() is constant and only required once here, making my comment absolutely obsolete.

Reply:
I think the compiler will evaluate this multiple times, so I can fix it.